lowranceusr.cc

```diff
@@ -1,238 +1,478 @@
-  { 10068, "Unknown Cache" },                 // ? mark
-  { 10045, "Locationless (Reverse) Cache" },  // hiding hunter
-  { 10066, "Post Office" },                   // letter
-  { 10019, "Event Cache" },                   // person
-  { 10070, "Webcam Cache" },                  // webcam
-  { 10042, "Disabled Cache" },                // black box with red X
-
-  {	 -1, nullptr }
+  { 10038,    "Geocache" },             // flag buoy
+  { 10016,    "Geocache Found" },       // exclamation
+  { 10043,    "Micro-Cache" },          // small red dot
+  { 10065,    "Virtual cache" },        // ghost
+  { 10051,    "Multi-Cache" },          // 4 dots in rhombus shape
+  { 10068,    "Unknown Cache" },        // ? mark
+  { 10045,    "Locationless (Reverse) Cache" },  // hiding hunter
+  { 10066,    "Post Office" },          // letter
+  { 10019,    "Event Cache" },          // person
+  { 10070,    "Webcam Cache" },         // webcam
+  { 10042,    "Disabled Cache" },       // black box with red X
+
+  // END OF ICON MAPPING
+  {    -1,    nullptr }
 };
 
-static gbfile* file_in;
-static gbfile* file_out;
-static short_handle mkshort_handle;
+static gbfile*        file_in;
+static gbfile*        file_out;
+static short_handle   mkshort_handle;
+
+static route_head*    trk_head;
+static route_head*    rte_head;
+
+static int            waypt_uid;
+static int            route_uid;
+static int            trail_uid;
+
+static char*          opt_ignoreicons;
+static char*          opt_writeasicons;
+static char*          opt_seg_break;
+static char*          opt_wversion;
+static char*          opt_rversion;
+static char*          opt_title;
+static char*          opt_content_descr;
+static char*          opt_serialnum;
+static int            opt_serialnum_i;
+
+static Waypoint**     waypt_table;
+static int            waypt_table_sz;
+static int            waypt_table_ct;
+
+/* from waypt.c, we need to iterate over waypoints when extracting routes */
+#if NEWQ
+extern QList<Waypoint*> waypt_list;
+#else
+extern queue          waypt_head;
+#endif
 
 static unsigned short waypt_out_count;
-static unsigned int trail_count, lowrance_route_count;
-static int trail_point_count;
-static char continuous = 1;
-static short num_section_points;
-static route_head* trk_head;
-static route_head* rte_head;
-static char* ignoreicons;
-static char* writeasicons;
-static char* merge;
-static char* seg_break;
-static char* wversion_arg;
-static int reading_version;
-static int writing_version;
+static int            trail_count, lowrance_route_count;
+static int            trail_point_count;
+static char           continuous = 1;
+static short          num_section_points;
+static char*          merge;
+static int            reading_version;
+static int            writing_version;
 
 #define MYNAME "Lowrance USR"
 
-#define MAXUSRSTRINGSIZE	256
-#define SEMIMINOR		   6356752.3142
-#define DEGREESTORADIANS	0.017453292
```

Comment:
I would still like to see this:
#define DEGREESTORADIANS (M_PI/180.0)
If not defined elsewhere defs.h will supply M_PI.
The compiler should do the division at compile time.
This will reduce the chance of reference file thrashing in the future as there isn't the temptation to add another digit.

Reply:
I was planning to make the change but I need to be able to spend enough time so I can make the change, regenerate the reference files, and then desk check to make sure I didn't mess anything up.  I am hoping to get that done in the next few days.

Reply:
Suggested changed adopted.